### ISSUE TITLE
can not connect to mysql when has special chars '@' in passwords

### ISSUE THREAD

**Issue by jagerzhang** [sqlalchemy/sqlalchemy#8327]
### Describe the bug

After the version is upgraded from `1.3.19` to `1.4.39`, if the password contains the @ symbol, the database connection will fail. it works well in  version `1.3.19`.

### To Reproduce

```bash
pip3 install sqlalchemy==1.4.39
```

```python
from sqlalchemy import create_engine
from sqlalchemy.orm import sessionmaker
from sqlalchemy.ext.declarative import declarative_base


def get_mysql_pool(user,
                   passwd,
                   host,
                   port,
                   database,
                   recycle_rate=1800,
                   pool_size=32,
                   max_overflow=64):
    declarative_base()
    sqlalchemy_database = f"mysql+pymysql://{user}:{passwd}@{host}:{port}/{database}"
    engine = create_engine(sqlalchemy_database,
                           pool_recycle=recycle_rate,
                           pool_size=pool_size,
                           max_overflow=max_overflow)
    return sessionmaker(autocommit=True, autoflush=True, bind=engine)

```

Change a password that contains an @ symbol, it will reproduce.

### Error

```
sqlalchemy.exc.OperationalError: (pymysql.err.OperationalError) (2003, "Can't connect to MySQL server on 'XXX@X.X.X.X' ([Errno -2] Name or service not known)")
``` 

### Versions

- OS: Centos 7.2
- Python: 3.6.8
- SQLAlchemy: 1.4.39
- Database:  MySQL
- DBAPI (eg: psycopg, cx_oracle, mysqlclient): mysqlclient

**Comment by CaselIT** [comment]
Hi. Special characters need to be escalated. See also https://github.com/sqlalchemy/sqlalchemy/issues?q=+label%3A%22%40+in+password%22+

**Comment by zzzeek** [comment]
this is getting ridiculous.  that it's always the @ sign, and never any of the other signs, points to the change we made.   Maybe we should just do a warning if more than one @ sign is in the URL or just raise.

**Comment by zzzeek** [comment]
opened #8328

**Comment by jagerzhang** [comment]
> Hi. Special characters need to be escalated. See also https://github.com/sqlalchemy/sqlalchemy/issues?q=+label%3A%22%40+in+password%22+
@zzzeek @CaselIT 

What I want to say is that this problem does not exist in version 1.3.19, indicating that there are versions after 1.3.19 that have deleted the special symbol compatibility

**Comment by zzzeek** [comment]
it's a fix that was in 1.4.16, so that database names with @ signs in them would parse correctly.  @ signs in passwords should never have worked.   we've just been waiting for months for people to stop upgrading and having this issue.

**Comment by jagerzhang** [comment]
> it's a fix that was in 1.4.16, so that database names with @ signs in them would parse correctly. @ signs in passwords should never have worked. we've just been waiting for months for people to stop upgrading and having this issue.

Well, I thought it was impossible to have the @ sign in the hostname.

**Comment by CaselIT** [comment]
> > Hi. Special characters need to be escalated. See also https://github.com/sqlalchemy/sqlalchemy/issues?q=+label%3A%22%40+in+password%22+
> > @zzzeek @CaselIT
> 
> Through version-by-version testing, it was found that this problem was introduced in version 1.4.16, and 1.4.15 did not have this problem.
> 
> > [engine] [bug] Fixed issue where an @ sign in the database portion of a URL would not
> > be interpreted correctly if the URL also had a username:password section.
> 
> #6482

Please note that the documentation has always said that special characters should be escalated
